apps/app/src/components/AddQuestionModal/QuestionEditor/QuestionEditor.tsx

```diff
@@ -0,0 +1,123 @@
+"use client";
+
+import { useEffect, useRef, useState, Suspense } from "react";
+import { twMerge } from "tailwind-merge";
+import dynamic from "next/dynamic";
+import BoldIcon from "../../../../public/icons/toolbar-bold.svg";
+import ItalicIcon from "../../../../public/icons/toolbar-italic.svg";
+import HeadingIcon from "../../../../public/icons/toolbar-heading.svg";
+import CodeIcon from "../../../../public/icons/toolbar-code.svg";
+import UlIcon from "../../../../public/icons/toolbar-ul.svg";
+import OlIcon from "../../../../public/icons/toolbar-ol.svg";
+import EyeIcon from "../../../../public/icons/toolbar-eye.svg";
+import { handleAction, Action } from "../../../lib/actions";
+import { ActionsGroup } from "./ActionsGroup";
+import { ActionItem } from "./ActionItem";
+
+const QuestionPreview = dynamic(
+	() => import("./QuestionPreview").then((mod) => mod.QuestionPreview),
+	{ ssr: false },
+);
+
+const actionIcons: Record<Action, JSX.Element> = {
+	BOLD: <BoldIcon viewBox="0 0 32 32" />,
+	ITALIC: <ItalicIcon viewBox="0 0 32 32" />,
+	HEADING: <HeadingIcon viewBox="0 0 24 24" />,
+	CODEBLOCK: <CodeIcon viewBox="0 0 32 32" />,
+	UL: <UlIcon viewBox="0 0 32 32" />,
+	OL: <OlIcon viewBox="0 0 32 32" />,
+};
+
+const hierarchy: Action[][] = [
+	["BOLD", "ITALIC", "HEADING"],
+	["CODEBLOCK", "UL", "OL"],
+];
+
+type QuestionEditorProps = Readonly<{
+	value: string;
+	onChange: (value: string) => void;
+}>;
+
+export const QuestionEditor = ({ value, onChange }: QuestionEditorProps) => {
+	const [isPreview, setIsPreview] = useState(false);
+
+	const textAreaRef = useRef<HTMLTextAreaElement | null>(null);
+	const selectionRef = useRef<{ start: number; end: number } | null>(null);
+
+	const handleActionClick = (action: Action) => () => {
+		if (!textAreaRef.current) {
+			return;
+		}
+
+		const { selectionStart, selectionEnd } = textAreaRef.current;
+		const { start, end, newValue } = handleAction({ selectionStart, selectionEnd, value, action });
```

Comment:
Uważam, że jest to nieco dziwne, że przekazujemy pola o nazwach selectionStart i selectionEnd, ale zwracamy już samo start i end.

Reply:
Moglibyśmy zrobić zawsze obiekt selection:{start, end}